services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/config/MongoDbConfigProperties.java

```diff
@@ -0,0 +1,270 @@
+     * string overrides the other connection settings. Format:
+     * mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
+     *
+     * @param connectionString connection string
+     * @return This instance for setter chaining.
+     */
+    public MongoDbConfigProperties setConnectionString(final String connectionString) {
+        this.connectionString = connectionString;
+        return this;
+    }
+
+    /**
+     * Gets the time in milliseconds that the mongo driver will wait to select a
+     * server for an operation before raising an error.
+     *
+     * @return time in milliseconds
+     */
+    public int getServerSelectionTimeout() {
+        return serverSelectionTimeoutMS;
+    }
+
+    /**
+     * Sets the time in milliseconds that the mongo driver will wait to select a
+     * server for an operation before raising an error.
+     *
+     * @param timeout timeout in milliseconds. Setting to zero means the default
+     *                value of Vert.x should be used.
+     * @return This instance for setter chaining.
+     */
+    public MongoDbConfigProperties setServerSelectionTimeout(final int timeout) {
+        this.serverSelectionTimeoutMS = timeout;
+        return this;
+    }
+
+    /**
+     * Gets the time in milliseconds to attempt a connection before timing out.
+     *
+     * @return time in milliseconds
+     */
+    public int getConnectTimeout() {
+        return connectTimeoutMS;
+    }
+
+    /**
+     * Sets the time in milliseconds to attempt a connection before timing out.
+     *
+     * @param connectTimeout timeout in milliseconds. Setting to zero means the default
+     *                       value of Vert.x should be used.
+     * @return This instance for setter chaining.
+     */
+    public MongoDbConfigProperties setConnectTimeout(final int connectTimeout) {
+        this.connectTimeoutMS = connectTimeout;
+        return this;
+    }
+
+    /**
+     * Gets the time in milliseconds to create indices during startup.
+     *
+     * @return time in milliseconds
+     */
+    public int getCreateIndicesTimeout() {
+        return createIndicesTimeoutMS;
+    }
+
+    /**
+     * Sets the time in milliseconds that the startup will try to create indices
+     * during startup.
+     *
+     * @param timeout timeout in milliseconds.
+     * @return This instance for setter chaining.
+     */
+    public MongoDbConfigProperties setCreateIndicesTimeout(final int timeout) {
+        this.createIndicesTimeoutMS = timeout;
+        return this;
+    }
+
+    /**
+     * Returns the properties of this instance in a JsonObject suitable for
+     * initializing a Vertx MongoClient object. Note: if the connectionString is
+     * set, it will override all other connection settings.
+     *
+     * @return MongoDB client config object
+     */
+    public JsonObject asMongoClientConfigJson() {
```

Comment:
the format returned by this method seems to be specific to the Mongo DB driver being actually used while the properties of this class seem to be generic. I would therefore propose to move this method to the `MongoDbExecutor` service which is already tied to the specific MongoDB driver ...